packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -35,6 +35,8 @@ library TaikoData {
         // The max number of L2 blocks that can stay unsynced on L1 (a value of zero disables
         // syncing)
         uint8 blockSyncThreshold;
+        // The number of L1 blocks after which a TKO snapshot should be taken.
+        uint32 tkoSnapshotInterval;
```

Comment:
Hmm, i'm not sure if this is a good way/practice to have snapshots taking at a predictable time. (Id ask Aragon maybe?)

If the tokens can be borrowed at a low interest rate OR a quick flash-loan can squeeze in right before a snapshot is taken, attacker can inflate his voting power with no (or low) extra cost.

Reply:
Stopped the auto merge. I do think we should talk to Aragon about this approach. 

Flashloan won't work as a way of stealing voting power as only a selected few addresses can take snapshot. But it is true that one can monitor pending propose-block transactions then buy a lot of taiko token, and sell them right after the snapshot is taken. However, people can always monitor snapshot transactions to front-run it to buy taiko tokens. No solution to my knowledge,so I do not see this is an issue that we need to solve. 

Let's discuss with Aragon before deciding if we should merge this PR.